src/crud.py

```
@@ -700,12 +923,30 @@ async def get_collection_by_id(
     )
     result = await db.execute(stmt)
     collection = result.scalar_one_or_none()
+    if collection is None:
+        logger.warning(f"Collection with ID '{collection_id}' not found for user {user_id}")
+        raise ResourceNotFoundException(f"Collection not found or does not belong to user")
```

Ruff (0.8.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)